### ISSUE TITLE
Changes to AccessibilityState fail to cause Narrator to read out the new state

### ISSUE THREAD

**Issue by aschultz** [microsoft/react-native-windows#8491]
Given a simple component like:
```
<View focusable={true} accessibilityRole={"checkbox"} accessibilityState={{ checked: isChecked }}>
    ...
</View>
```
Changing the value of 'checked' is not triggering Narrator to read out the new state.

Best I can tell, RNW should be calling `AutomationPeer::RaisePropertyChangedEvent` whenever these values change. There are a few examples of this in the WinUI codebase like this:

```
if (auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*this))
        {
            const auto newValue = IsChecked() ? winrt::ToggleState::On : winrt::ToggleState::Off;
            const auto oldValue = (newValue == winrt::ToggleState::On) ? winrt::ToggleState::Off : winrt::ToggleState::On;
            peer.RaisePropertyChangedEvent(winrt::TogglePatternIdentifiers::ToggleStateProperty(), box_value(oldValue), box_value(newValue));
        }
```
